### PR TITLE
Remove checks that compare this with NULL in baseAST.cpp and expr.cpp

### PR DESCRIPTION
**Pull request by noakesmichael**

### CODE REVIEW

File: compiler/passes/createTaskFunctions.cpp [hunk 1]
```diff
@@ -485,7 +485,9 @@ void createTaskFunctions(void) {
 
           markOuterVarsWithIntents(block->byrefVars, uses);
```
Comment (Kyle-B) [inline compiler/passes/createTaskFunctions.cpp]: All 3 of these lines might be able to not run if `block->byrefVars` is null. Your call if you want to find out.

**Comment by Kyle-B** [comment]
Looks fine to me.

**Comment by noakesmichael** [comment]
Thx

Mike

On Apr 13, 2015, at 6:34 PM, Kyle Brady <notifications@github.com<mailto:notifications@github.com>> wrote:


Looks fine to me.

**Comment by mppf** [comment]
Looks good, thanks for fixing.

**Pull request merged**